Segfault during shutdown

Seems like a1e969ddc178a2865509f07714124c4a32763baa introduced a regression that makes mke segfault on shutdown:

```
INFO[2020-10-06 12:39:36] Shutting down pid 399061                      component=kube-controller-manager
INFO[2020-10-06 12:39:36] lost leader lease                             component=applier-manager
E1006 12:39:36.910051  399032 runtime.go:78] Observed a panic: "send on closed channel" (send on closed channel)
goroutine 178 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x10db820, 0x14e5180)
        /home/ncopa/go/pkg/mod/k8s.io/apimachinery@v0.19.1/pkg/util/runtime/runtime.go:74 +0x84
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /home/ncopa/go/pkg/mod/k8s.io/apimachinery@v0.19.1/pkg/util/runtime/runtime.go:48 +0x7c
panic(0x10db820, 0x14e5180)
        /usr/lib/go-1.15/src/runtime/panic.go:969 +0x11c
github.com/Mirantis/mke/pkg/applier.(*Manager).electLeader.func2()
        /home/ncopa/go/src/github.com/Mirantis/mke/pkg/applier/manager.go:123 +0x80
k8s.io/client-go/tools/leaderelection.(*LeaderElector).Run.func1(0x400070eea0)
        /home/ncopa/go/pkg/mod/k8s.io/client-go@v0.19.1/tools/leaderelection/leaderelection.go:200 +0x2c
k8s.io/client-go/tools/leaderelection.(*LeaderElector).Run(0x400070eea0, 0x153ebe0, 0x40001d4080)
        /home/ncopa/go/pkg/mod/k8s.io/client-go@v0.19.1/tools/leaderelection/leaderelection.go:204 +0x170
github.com/Mirantis/mke/pkg/applier.(*Manager).electLeader(0x40004481b0, 0x12f8d29, 0x14, 0x12e782a, 0xf, 0x40004f1b00, 0x40, 0x0, 0x0)
        /home/ncopa/go/src/github.com/Mirantis/mke/pkg/applier/manager.go:140 +0x2a0
created by github.com/Mirantis/mke/pkg/applier.(*Manager).Run
        /home/ncopa/go/src/github.com/Mirantis/mke/pkg/applier/manager.go:80 +0x1c8
panic: send on closed channel [recovered]
        panic: send on closed channel

goroutine 178 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /home/ncopa/go/pkg/mod/k8s.io/apimachinery@v0.19.1/pkg/util/runtime/runtime.go:55 +0x104
panic(0x10db820, 0x14e5180)
        /usr/lib/go-1.15/src/runtime/panic.go:969 +0x11c
github.com/Mirantis/mke/pkg/applier.(*Manager).electLeader.func2()
        /home/ncopa/go/src/github.com/Mirantis/mke/pkg/applier/manager.go:123 +0x80
k8s.io/client-go/tools/leaderelection.(*LeaderElector).Run.func1(0x400070eea0)
        /home/ncopa/go/pkg/mod/k8s.io/client-go@v0.19.1/tools/leaderelection/leaderelection.go:200 +0x2c
k8s.io/client-go/tools/leaderelection.(*LeaderElector).Run(0x400070eea0, 0x153ebe0, 0x40001d4080)
        /home/ncopa/go/pkg/mod/k8s.io/client-go@v0.19.1/tools/leaderelection/leaderelection.go:204 +0x170
github.com/Mirantis/mke/pkg/applier.(*Manager).electLeader(0x40004481b0, 0x12f8d29, 0x14, 0x12e782a, 0xf, 0x40004f1b00, 0x40, 0x0, 0x0)
        /home/ncopa/go/src/github.com/Mirantis/mke/pkg/applier/manager.go:140 +0x2a0
created by github.com/Mirantis/mke/pkg/applier.(*Manager).Run
        /home/ncopa/go/src/github.com/Mirantis/mke/pkg/applier/manager.go:80 +0x1c8
```

--- a/pkg/applier/manager.go
+++ b/pkg/applier/manager.go
@@ -55,10 +55,6 @@ func (m *Manager) retrieveKubeClient() error {
 func (m *Manager) Run() error {
 	log := m.log
 
-	// Make the done channels
-	m.tickerDone = make(chan struct{})
-	m.watcherDone = make(chan struct{})
-
 	for m.client == nil {
 		log.Debug("retrieving kube client config")
 		_ = m.retrieveKubeClient()
@@ -87,8 +83,6 @@ func (m *Manager) Run() error {
 
 // Stop stops the Manager
 func (m *Manager) Stop() error {
-	close(m.tickerDone)
-	close(m.watcherDone)
 	m.cancelLeaderElection()
 	return nil
 }
@@ -103,12 +97,14 @@ func (m *Manager) watchLeaseEvents(events *leaderelection.LeaseEvents) {
 			changesDetected := &atomic.Value{}
 			// to make first tick to sync everything and retry until it succeeds
 			changesDetected.Store(true)
+			m.tickerDone = make(chan struct{})
+			m.watcherDone = make(chan struct{})
 			go m.runFSWatcher(changesDetected)
 			go m.runApplier(changesDetected)
 		case <-events.LostLease:
 			log.Info("lost leader lease")
-			m.tickerDone <- struct{}{}
-			m.watcherDone <- struct{}{}
+			close(m.tickerDone)
+			close(m.watcherDone)
 		}
 	}
 }

